Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,299 @@
+
+        // Create policy with 1 day read permission
+        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
+        EnumSet<SharedAccessBlobPermissions> perEnumSet = EnumSet.of(SharedAccessBlobPermissions.READ);
+        policy.setPermissions(perEnumSet);
+
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.DATE, 1);
+        policy.setSharedAccessExpiryTime(c.getTime());
+
+        // Create SAS key
+        String sas = blob.generateSharedAccessSignature(policy, null);
+        return blob.getUri() + "?" + sas;
+    }
+
+    // Create a job with a single task
+    private static void submitJobAndAddTask(BatchClient client, CloudBlobContainer container, String poolId, String jobId) throws BatchErrorException, IOException, StorageException, InvalidKeyException, URISyntaxException {
+        String FILE_NAME = "mytest.txt";
+
+        // Create job run at the specified pool
+        PoolInformation poolInfo = new PoolInformation();
+        poolInfo.setPoolId(poolId);
+        client.getJobOperations().createJob(jobId, poolInfo);
+
+        // Create task
+        TaskAddParameter taskToAdd = new TaskAddParameter();
+        taskToAdd.setId("mytask");
+        taskToAdd.setCommandLine(String.format("cat %s", FILE_NAME));
+
+        String sas = uploadFileToCloud(container);
+
+        // Associate resource file with task
+        ResourceFile file = new ResourceFile();
+        file.setFilePath(FILE_NAME);
+        file.setBlobSource(sas);
+        List<ResourceFile> files = new ArrayList<ResourceFile>();
+        files.add(file);
+        taskToAdd.setResourceFiles(files);
+
+        // Add task to job
+        client.getTaskOperations().createTask(jobId, taskToAdd);
+    }
+
+    // Wait all tasks under a specified job to be completed
+    private static boolean waitForTasksToComplete(BatchClient client, String jobId, Duration expiryTime) throws BatchErrorException, IOException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+
+        while (elapsedTime < expiryTime.toMillis()) {
+            List<CloudTask> taskCollection = client.getTaskOperations().listTasks(jobId, new DetailLevel.Builder().selectClause("id, state").build());
+
+            boolean allComplete = true;
+            for (CloudTask task : taskCollection) {
+                if (task.getState() != TaskState.COMPLETED) {
+                    allComplete = false;
+                    break;
+                }
+            }
+
+            if (allComplete) {
+                // All tasks completed
+                return true;
+            }
+
+            System.out.println("wait 10 seconds for tasks to complete...");
+
+            // Check again after 10 seconds
+            Thread.sleep(10 * 1000);
+            elapsedTime = (new Date()).getTime() - startTime;
+        }
+
+        // Timeout, return false
+        return false;
```

Comment:
Does timeout case make more sense to throw and exception, and make this not boolean but void instead?